exporter.go

```diff
@@ -115,9 +115,18 @@ func (e *Exporter) Export(opts ExportOptions) (files.Report, error) {
 	meta.RunImage.Reference = opts.RunImageRef
 
 	if e.PlatformAPI.AtLeast("0.12") {
+		// if it's new API version but only using Stack still, then normalize the model by copying stack into the RunImage
+		if opts.RunImageForExport.Image == "" && len(opts.RunImageForExport.Mirrors) == 0 {
+			opts.RunImageForExport.Image = opts.Stack.RunImage.Image
+			opts.RunImageForExport.Mirrors = opts.Stack.RunImage.Mirrors
+		}
 		meta.RunImage.Image = opts.RunImageForExport.Image
 		meta.RunImage.Mirrors = opts.RunImageForExport.Mirrors
-	} else {
+		// ensure we always copy the new RunImage into the old stack to preserve old behavior
+		meta.Stack = &files.Stack{}
+		meta.Stack.RunImage.Image = meta.RunImage.Image
+		meta.Stack.RunImage.Mirrors = meta.RunImage.Mirrors
```

Comment:
```suggestion
		meta.Stack = opts.Stack
```
This should be equivalent because of https://github.com/buildpacks/lifecycle/blob/28084b74124ad4484bd9e67ea4c469b05b5f38fa/cmd/lifecycle/exporter.go#L219